src/basic/mountpoint-util.c

```diff
@@ -849,3 +849,36 @@ bool path_below_api_vfs(const char *p) {
         /* API VFS are either directly mounted on any of these three paths, or below it. */
         return PATH_STARTSWITH_SET(p, "/dev", "/sys", "/proc");
 }
+
+int find_mountpoint(const char *device_path, char **ret_mountpoint) {
+
+        _cleanup_fclose_ FILE *fp = NULL;
+        fp = fopen("/proc/mounts", "r");
```

Comment:
"re", not just "r" in our codebase please.

but, uh, /proc/mounts is obsolete (/proc/self/mountinfo), and please do not add new parses for this, given that there's already two in glibc and libmount. We actually try to unify things on the latter.